File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-jamba-development/Android_Raleys/Raleys/src/com/raleys/app/android/models/Offer.java
package com.raleys.app.android.models;

import java.io.Serializable;
import java.util.ArrayList;

public class Offer implements Serializable {
	private static final long serialVersionUID = 8714970475455405873L;
	public String offerType;
	public String offerProductImage;
	public String consumerDesc;
	public String consumerDetails;
	public String offerSize;
	public String offerLimit;
	public String offerId;
	public String offeridLive;
	public String offerProductImageFile;
	public float offerPrice;
	public String offerLogoImageFile;
	public String acceptGroup;
	public String consumerTitle;
	public String offerOrderNo;
	public String offerListType;
	public String startDate;
	public String offerProductImageAlt;
	public String candidateGroup;
	public String offerLogoImageAlt;
	public String offerLogoImage;
	public String active;
	public String hideAccept;
	public String endDate;
	public ArrayList<Offer> offerList;

	public boolean _acceptableOffer; // not used by json
	public boolean _acceptedOffer; // not used by json

	public String promoCode;
	public boolean dynamicOffer;// newly added for closed loop fix
}

File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-Jamba-Android-v2.4.1/iOS_Raleys/Raley's/Raley's/Classes/Models/Offer.m
//
//  Offer.m
//  Raley's
//
//  Created by Billy Lewis on 10/3/13.
//  Copyright (c) 2013 Raleys. All rights reserved.
//

#import "Offer.h"

@implementation Offer

@synthesize offerType;
@synthesize offerProductImage;
@synthesize consumerDesc;
@synthesize consumerDetails;
@synthesize offerSize;
@synthesize offerLimit;
@synthesize offerId;
@synthesize offeridLive;
@synthesize offerProductImageFile;
@synthesize offerPrice;
@synthesize offerLogoImageFile;
@synthesize acceptGroup;
@synthesize consumerTitle;
@synthesize offerOrderNumber;
@synthesize offerListType;
@synthesize startDate;
@synthesize offerProductImageAlt;
@synthesize candidateGroup;
@synthesize offerLogoImageAlt;
@synthesize offerLogoImage;
@synthesize active;
@synthesize hideAccept;
@synthesize endDate;
@synthesize offerList;
@synthesize _acceptedOffer;
@synthesize _acceptableOffer;
@synthesize promoCode;

- (NSString *)getComponentTypeForArray:(NSString *)propertyName
{
    if([propertyName isEqualToString:@"offerList"])
        return @"Offer";
    else
        return nil;
}

@end
